data/transactions/verify/txn.go

```diff
@@ -448,3 +456,394 @@ func (w *worksetBuilder) next() (txnGroups [][]transactions.SignedTxn) {
+				// no point in blocking and waiting for a seat here, let it wait in the exec pool
+				err := sv.addVerificationTaskToThePool(makeSingleUnverifiedElement(sv.nbw, sv.ledger, stx), false)
+				if err != nil {
+					continue // pool is terminated. nothing to report to the txn sender
+				}
+				continue // stx is handled, continue
+			}
+
+			// add this txngrp to the list of batchable txn groups
+			numberOfSigsInCurrent = numberOfSigsInCurrent + numberOfBatchableSigsInGroup
+			uel.elementList = append(uel.elementList, stx)
+			if numberOfSigsInCurrent > txnPerWorksetThreshold {
+				// enough transaction in the batch to efficiently verify
+
+				if numberOfSigsInCurrent > 4*txnPerWorksetThreshold {
+					// do not consider adding more txns to this batch.
+					// bypass the seat count and block if the exec pool is busy
+					// this is to prevent creation of very large batches
+					err := sv.addVerificationTaskToThePool(uel, false)
+					if err != nil {
+						continue // pool is terminated.
+					}
+					added = true
+				} else {
+					added = sv.processBatch(uel)
+				}
+				if added {
+					numberOfSigsInCurrent = 0
+					uel = makeUnverifiedElementList(sv.nbw, sv.ledger)
+					// starting a new batch. Can wait long, since nothing is blocked
+					timer.Reset(waitForFirstTxnDuration)
+				} else {
+					// was not added because no-available-seats. wait for some more txns
+					timer.Reset(waitForNextTxnDuration)
+				}
+			} else {
+				if isFirstInBatch {
+					// an element is added and is waiting. shorten the waiting time
+					timer.Reset(waitForNextTxnDuration)
+				}
+			}
+		case <-timer.C:
+			// timer ticked. it is time to send the batch even if it is not full
+			if numberOfSigsInCurrent == 0 {
+				// nothing batched yet... wait some more
+				timer.Reset(waitForFirstTxnDuration)
+				continue
+			}
+			added = sv.processBatch(uel)
+			if added {
+				numberOfSigsInCurrent = 0
+				uel = makeUnverifiedElementList(sv.nbw, sv.ledger)
+				// starting a new batch. Can wait long, since nothing is blocked
+				timer.Reset(waitForFirstTxnDuration)
+			} else {
+				// was not added because no-available-seats. wait for some more txns
+				timer.Reset(waitForNextTxnDuration)
+			}
+		case <-sv.ctx.Done():
+			if numberOfSigsInCurrent > 0 {
+				// send the current accumulated transactions to the pool
+				err := sv.addVerificationTaskToThePool(uel, false)
+				if err != nil {
+					continue // pool is terminated.
+				}
+			}
+			// wait for the pending tasks, then close the result chan
+			sv.pendingTasksWg.Wait()
+			return
+		}
+	}
+}
+
+func (sv *StreamVerifier) sendResult(veTxnGroup []transactions.SignedTxn, veContext interface{}, err error) {
+	vr := VerificationResult{
+		TxnGroup: veTxnGroup,
+		Context:  veContext,
+		Err:      err,
+	}
+	// send the txn result out the pipe
+	// this should never block. the receiver end of this channel will drop transactions if the
+	// postVerificationQueue is blocked
```

Comment:
So when this operation is blocked, the execpool will get blocked, and the system may deadlock. 
in txHandler.go, where this channel is consumed, we have: 
```		case result := <-handler.streamReturnChan:
			txBLMsg := result.Context.(*txBacklogMsg)
			txBLMsg.verificationErr = result.Err
			select {
			case handler.postVerificationQueue <- txBLMsg:
			default:
				// we failed to write to the output queue, since the queue was full.
				// adding the metric here allows us to monitor how frequently it happens.
				transactionMessagesDroppedFromPool.Inc(nil)
			} 
```
`sv.resultChan` is `handler.streamReturnChan`, and as you can see here, the only operation that might get blocked here is passing the element to `handler.postVerificationQueue`. But it will not block, because if the `handler.postVerificationQueue` buffer is full, then the message will be dropped and reported.